instance/gandi.py

```diff
@@ -61,15 +64,38 @@ def client_zone(self):
         """
         return self.client.domain.zone
 
+    def _populate_zone_id_cache(self):
+        """
+        Populate the internal zone ID cache with all domains in the current Gandi account.
+        """
+        self._zone_id_cache = {}
+        for domain_dict in self.client.domain.list(self.api_key):
+            domain = domain_dict['fqdn'].lower()
+            self._zone_id_cache[domain] = self.client.domain.info(self.api_key, domain)['zone_id']
+
+    def split_domain_name(self, domain):
+        """
+        Split the given domain name in the registered domain and the subdomain.
+        """
+        if self._zone_id_cache is None:
+            self._populate_zone_id_cache()
+        labels = domain.lower().split('.')
+        for split_index in range(len(labels) - 1):
+            registered_domain = '.'.join(labels[split_index:])
+            if registered_domain in self._zone_id_cache:
+                subdomain = '.'.join(labels[:split_index]) or '@'
+                return subdomain, registered_domain
+        raise ValueError(
+            'The given domain name does not match any domain registered in the Gandi account.'
+        )
```

Comment:
It should show up in the instance log already.  (Not in the appserver log, since the DNS record is set before a new appserver is even created, so there is no appserver to log it to.)